### ISSUE TITLE
add universe level inference

### ISSUE THREAD

**Issue by stchang** [wilbowma/cur#136]

**Comment by wilbowma** [comment]
I don't know how to do this locally. The algorithm Coq uses is a module-level constraint solver, which I disapprove of :(

**Comment by stchang** [comment]
yeah i agree. just remembered it was missing and couldnt find an issue for it
